### DISCUSSION TITLE
VR for left handed

### DISCUSSION THREAD

**Discussion by GiftOfHell** [Okabintaro/SubmersedVR#142] [Q&A]
I want to play subnautica in vr, but handling with everything with my right hand is pretty hard. Is it possible to change hands? 

(like, I need at least to know is it worth it to start editing mod or ets to achive the goal)

**Comment by Okabintaro** [comment]
Yes, it's possible to change hands in SubmersedVR. The mod uses SteamVR's input system, allowing you to modify control bindings. You can reassign controls between hands, including swapping thumbstick functions. Before diving into mod editing, try adjusting the SteamVR bindings first. This guide explains the process (though for another game, the principles are the same): https://steamcommunity.com/sharedfiles/filedetails/?id=2029205314

**Reply by farmboy0** [reply]
The problem is the PDA which is inherently right handed. The mod would need to change the app to hold the PDA with the right hand so you could use it with the pointer in the left hand. Besides this all tools are held with the right hand. You'd need to change all that to hold the tools with the left hand.

**Reply by Okabintaro** [reply]
Ah I am afraid that's a bit harder to do, since the game hardcodes the position of the pda and tools.
You could try just swapping the tracked controllers here: https://github.com/Okabintaro/SubmersedVR/blob/master/SubmersedVR/VR/VRCameraRig.cs#L118-L127
But I am not sure it will work nicely. I think it could be fine since we don't render the body(otherwise you would see the arms twisted).

**Reply by farmboy0** [reply]
That wont look nice but should be doable. Do you think its possible to attach tools to the left hand and the PDA to the right hand with this mod?

**Reply by Okabintaro** [reply]
Not yet. I have no idea how hard it would be to be honest. The tool attachement is in built into the games code. It's probably quite a bit of work to figure it out.

**Reply by GiftOfHell** [reply]
I found a way to switch hands & laser, and mirrored hands. With controls change, it is quite perfect, even PDA works well, but the only problem left is that hands are turned inside out. It'll be cool if you could suggest a way to fix this.

**Comment by farmboy0** [comment]
@GiftOfHell do you have something to check out?

**Reply by GiftOfHell** [reply]
I made fork right now, but failed something... 
Too many changes, so:

https://github.com/GiftOfHell/SubmersedVR_LeftHand

I changed only:
SubmersedVR/VR/VRCameraRig.cs
Switched hands on 114/117 and 148/151 lines

SubmersedVR/VR/VRHands.cs
All coords, 154 and method on 230 to correct hand coords

I thought of turnig inside out hands somewhere in VRHands.cs, but haven't found a way.
